Automattic/woocommerce-payments
# Integrate product and cart pages with Payment Request Button

Fixes https://github.com/Automattic/woocommerce-payments/issues/1072

#### Changes proposed in this Pull Request

Copies (https://github.com/Automattic/woocommerce-payments/commit/ee5a56e93b8993b0cf5532bb2634c640bf356e78) and adapts the Payment Request integration code from the Stripe extension [[PHP](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/master/includes/payment-methods/class-wc-stripe-payment-request.php), [JS](https://github.com/woocommerce/woocommerce-gateway-stripe/blob/master/assets/js/stripe-payment-request.js)] to the context of WCPay – the goal is to provide identical functionality to the existing battle-hardened integration.

All available tests were copied and adapted from the Stripe Gateway extension. The client doesn't have tests yet, but we plan to sync here once we add new tests in the Stripe repository.

#### Testing instructions



For the purpose of this test, you can test either Google Pay, using Chrome, or Microsoft Pay, if you use Windows.

For Apple Pay, the domain needs to be registered with Apple first. #1239 will take care of that.

- Make sure you have an HTTPS URL for your test setup.
- Go through the onboarding process or get yourself connected with an existing Stripe express account.
- Go to the Payments > Settings page.
- Make sure you have "Enable WooCommerce Payments" and "Enable Payment Request Button" checked.
- Open your HTTPS URL in Chrome and navigate to a product page. If you have a saved card in [chrome://settings/payments](chrome://settings/payments) you should be able to see the Payment Request Button.
- Try editing the Payment Request Button settings (type, theme, label and height) and check if the button changes accordingly.
- Make sure "Test mode" is enabled in settings and try making a purchase using the Payment Request Button.

#### Screenshots

Chrome - Branded - Text and logo

<img width="423" alt="Screen Shot 2021-01-30 at 12 39 40" src="https://user-images.githubusercontent.com/5509901/106361170-404c7d00-62fb-11eb-9369-dc9fdfd1641d.png">

Chrome - Branded - Logo only

<img width="421" alt="Screen Shot 2021-01-30 at 12 40 15" src="https://user-images.githubusercontent.com/5509901/106361189-5fe3a580-62fb-11eb-95f4-d5af483cd1ce.png">

Default - Buy - Dark

<img width="423" alt="Screen Shot 2021-01-30 at 12 45 01" src="https://user-images.githubusercontent.com/5509901/106361212-7e49a100-62fb-11eb-809d-55686ad22f73.png">

Default - Donate - Dark

<img width="423" alt="Screen Shot 2021-01-30 at 12 44 36" src="https://user-images.githubusercontent.com/5509901/106361216-84d81880-62fb-11eb-882c-98be12a0cfc2.png">

Default - Buy - Light

<img width="423" alt="Screen Shot 2021-01-30 at 12 45 22" src="https://user-images.githubusercontent.com/5509901/106361222-902b4400-62fb-11eb-8930-c833e05a61ea.png">

Default - Buy - Light-Outline

<img width="428" alt="Screen Shot 2021-01-30 at 12 45 56" src="https://user-images.githubusercontent.com/5509901/106361245-a802c800-62fb-11eb-9a2d-90d1abd982dd.png">

Custom

<img width="425" alt="Screen Shot 2021-01-30 at 12 49 07" src="https://user-images.githubusercontent.com/5509901/106361367-4ee76400-62fc-11eb-932c-9a68a3fe479d.png">

On Microsoft Edge it defaults to Stripe's default button style.

![edge](https://user-images.githubusercontent.com/5509901/106361152-23b04500-62fb-11eb-93dc-a396042e616b.jpeg)

-------------------
